core/constant/common.go

```diff
@@ -40,4 +40,105 @@ const (
 	FilePerm = 0644
 )
 
+var WebUrlMap = map[string]struct{}{
+	"/apps":           {},
+	"/apps/all":       {},
+	"/apps/installed": {},
+	"/apps/upgrade":   {},
+
+	"/containers":           {},
+	"/containers/container": {},
+	"/containers/image":     {},
+	"/containers/network":   {},
+	"/containers/volume":    {},
+	"/containers/repo":      {},
+	"/containers/compose":   {},
+	"/containers/template":  {},
+	"/containers/setting":   {},
+
+	"/cronjobs": {},
+
+	"/databases":                   {},
+	"/databases/mysql":             {},
+	"/databases/mysql/remote":      {},
+	"/databases/postgresql":        {},
+	"/databases/postgresql/remote": {},
+	"/databases/redis":             {},
+	"/databases/redis/remote":      {},
+
+	"/hosts":                  {},
+	"/hosts/files":            {},
+	"/hosts/monitor/monitor":  {},
+	"/hosts/monitor/setting":  {},
+	"/hosts/terminal":         {},
+	"/hosts/firewall/port":    {},
+	"/hosts/firewall/forward": {},
+	"/hosts/firewall/ip":      {},
+	"/hosts/process/process":  {},
+	"/hosts/process/network":  {},
+	"/hosts/ssh/ssh":          {},
+	"/hosts/ssh/log":          {},
+	"/hosts/ssh/session":      {},
+
+	"/logs":           {},
+	"/logs/operation": {},
+	"/logs/login":     {},
+	"/logs/website":   {},
+	"/logs/system":    {},
+	"/logs/ssh":       {},
+
+	"/settings":               {},
+	"/settings/panel":         {},
+	"/settings/backupaccount": {},
+	"/settings/license":       {},
+	"/settings/about":         {},
+	"/settings/safe":          {},
+	"/settings/snapshot":      {},
+	"/settings/expired":       {},
+
+	"/toolbox":              {},
+	"/toolbox/device":       {},
+	"/toolbox/supervisor":   {},
+	"/toolbox/clam":         {},
+	"/toolbox/clam/setting": {},
+	"/toolbox/ftp":          {},
+	"/toolbox/fail2ban":     {},
+	"/toolbox/clean":        {},
+
+	"/websites":                 {},
+	"/websites/ssl":             {},
+	"/websites/runtimes/php":    {},
+	"/websites/runtimes/node":   {},
+	"/websites/runtimes/java":   {},
+	"/websites/runtimes/go":     {},
+	"/websites/runtimes/python": {},
+	"/websites/runtimes/dotnet": {},
+
+	"/login": {},
+
+	"/xpack":                   {},
+	"/xpack/waf/dashboard":     {},
+	"/xpack/waf/global":        {},
+	"/xpack/waf/websites":      {},
+	"/xpack/waf/log":           {},
+	"/xpack/waf/block":         {},
+	"/xpack/monitor/dashboard": {},
+	"/xpack/monitor/setting":   {},
+	"/xpack/monitor/rank":      {},
+	"/xpack/monitor/log":       {},
+	"/xpack/tamper":            {},
+	"/xpack/gpu":               {},
+	"/xpack/alert/dashboard":   {},
+	"/xpack/alert/log":         {},
+	"/xpack/alert/setting":     {},
+	"/xpack/setting":           {},
+}
+
+var DynamicRoutes = []string{
+	`^/containers/composeDetail/[^/]+$`,
+	`^/databases/mysql/setting/[^/]+/[^/]+$`,
+	`^/databases/postgresql/setting/[^/]+/[^/]+$`,
+	`^/websites/[^/]+/config/[^/]+$`,
+}
+
 var CertStore atomic.Value
```

Comment:
The provided code snippet appears to be part of an application configuration that manages various endpoints URLs. It seems there are some minor inconsistencies between the original and suggested versions, particularly concerning variable names.

For example:

- The var declaration `cert store` is missing from `web/url map`.
- The dynamic route definition has a typo (the second route should have `"setting"` instead of `"s"").
- There might also be a typographical error in the web url mapping.
- "dynamic routes" should likely use capitalized letters (`DynamicRoutes`).

However, without seeing any specific discrepancies or actual problems with these definitions, it would be challenging to provide detailed suggestions about how to optimize this section of the code. Nonetheless, here's what I've found out based on my understanding at present:

1) **URL Mapping**: Ensure all URL mappings are spelled correctly, and they do not overlap too much. Also ensure no leading slashes (`^/`) if your patterns start on the next line, because otherwise it could lead to confusion.

2) **Variable Names**: Consistently capitalize variables' names for better readability when working with values coming back from database operations. 

3) **Path Matching**: Pay close attention to path matching logic in each pattern. This can help keep URLs consistent in terms of structure across different endpoint categories but avoids unnecessary parsing for simple cases like trailing slash (/).

Please review the documentation comments for any other potential issues you need guidance on.